src/Options/Applicative/Help/Types.hs

```diff
@@ -42,6 +48,43 @@ helpText (ParserHelp e s h u d b g f) =
 -- | Convert a help text to 'String'.
 renderHelp :: Int -> ParserHelp -> String
 renderHelp cols
-  = (`displayS` "")
-  . renderPretty 1.0 cols
+  =  (`renderShowS` "")
+  . renderAnsi
+  . layoutPretty (LayoutOptions (AvailablePerLine cols 1.0))
   . helpText
+
+renderAnsi :: SimpleDocStream Ann -> SimpleDocStream ()
+renderAnsi
+  = renderCtxDecorated defaultStyle renderPush renderPop
+  . alterAnnotationsS alter
+  where
+    alter :: Ann -> Maybe SetStyle
+    alter (AnnStyle setStyle) = Just setStyle
+    renderPush :: SetStyle -> SetStyle -> SimpleDocStream () -> SimpleDocStream ()
+    renderPush _ setStyle = SText 0 (fromString (styleToRawText setStyle))
+    renderPop :: SetStyle -> SetStyle -> SimpleDocStream () -> SimpleDocStream ()
+    renderPop setStyle _ = SText 0 (fromString (styleToRawText setStyle))
+
+renderCtxDecorated
+    :: forall ann.
+       Monoid ann
+    => ann
+    -> (ann -> ann -> SimpleDocStream () -> SimpleDocStream ())  -- ^ How to render an annotation
+    -> (ann -> ann -> SimpleDocStream () -> SimpleDocStream ())  -- ^ How to render the removed annotation
+    -> SimpleDocStream ann
+    -> SimpleDocStream ()
+renderCtxDecorated topAnn push pop = go [topAnn]
+  where
+    go :: [ann] -> SimpleDocStream ann -> SimpleDocStream ()
+    go _                      SFail               = SFail
+    go []                     SEmpty              = SEmpty
+    go (_:_:_)                SEmpty              = SEmpty
+    go (_:_)                  SEmpty              = SEmpty
+    go stack                  (SChar c rest)      = SChar c (go stack rest)
+    go stack                  (SText l t rest)    = SText l t (go stack rest)
+    go stack                  (SLine i rest)      = SLine i (go stack rest)
+    go stack@(ctxAnn:_)       (SAnnPush ann rest) = push ctxAnn ann (go ((ctxAnn <> ann) : stack) rest)
+    go (ann:stack@(ctxAnn:_)) (SAnnPop rest)      = pop ctxAnn ann (go stack rest)
+    go _                      (SAnnPush _ _)      = error "An unpaired style initiator was encountered. This is a bug in the layout algorithm.  Please report this as a bug"
+    go _                      (SAnnPop _)         = error "An unpaired style terminator was encountered. This is a bug in the layout algorithm.  Please report this as a bug"
+{-# INLINE renderCtxDecorated #-}
```

Comment:
This code has been updated with @Bodigrim 's suggestion to avoid using `text` and it looks a bit simpler.